op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -417,3 +397,65 @@ func openCeloDb(path string) (ethdb.Database, error) {
 	}
 	return ldb, nil
 }
+
+// addAllocsToCeloState adds the allocated accounts to the Celo state database.
+// It iterates through the accounts in the provided `genesis` struct and sets them inside the `db` state database.
+// This will change the state root
+// Another property is that the total balance changes must be 0
+func addAllocsToCeloState(db *state.StateDB, genesis *core.Genesis) error {
+	log.Info("Migrating OP contracts into state DB", "num_allocs", len(genesis.Alloc))
+
+	accountCounter := 0
+	overwriteCounter := 0
+	for k, v := range genesis.Alloc {
+		accountCounter++
+		if db.Exist(k) {
+			equal := bytes.Equal(db.GetCode(k), v.Code)
+
+			log.Warn("Operating on existing state", "account", k, "equalCode", equal)
+			overwriteCounter++
+		}
+		// TODO(pl): decide what to do with existing accounts.
+		db.CreateAccount(k)
+
+		// CreateAccount above copied the balance, check if we change it
+		if db.GetBalance(k).Cmp(uint256.MustFromBig(v.Balance)) != 0 {
+			// TODO(pl): make this a hard error once the migration has been tested more
+			log.Warn("Moving account changed native balance", "address", k, "oldBalance", db.GetBalance(k), "newBalance", v.Balance)
+		}
+
+		db.SetNonce(k, v.Nonce)
+		db.SetBalance(k, uint256.MustFromBig(v.Balance))
+		db.SetCode(k, v.Code)
+		db.SetStorage(k, v.Storage)
+
+		log.Info("Moved account", "address", k)
+	}
+	log.Info("Migrated OP contracts into state DB", "copiedAccounts", accountCounter, "overwrittenAccounts", overwriteCounter)
+	return nil
+}
+
+func migrateTestnetAccounts(db *state.StateDB, config *params.ChainConfig) error {
+	if mapping, exists := contractMigrations[config.ChainID.Uint64()]; exists {
+		log.Info("Found contract migrations for chain", "chainID", config.ChainID, "mappings", len(mapping))
+		for source, target := range mapping {
+			if !db.Exist(source) {
+				log.Warn("Source account does not exist", "source", source)
```

Comment:
Should this be a higher severity error? It seems like if the source doesn't exist something is probably wrong with the data or the script right?